crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -628,16 +632,49 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
 
         let customer_id = payment_data.payment_intent.customer_id.clone();
 
+        let raw_customer_details = if let Some(customer) = customer {
+            Some(CustomerData {
+                name: customer.name.map(|name| name.into_inner()).clone(),
+                email: customer.email.map(Email::from).clone(),
+                phone: customer.phone.map(|phone| phone.into_inner()).clone(),
+                phone_country_code: customer.phone_country_code.clone(),
+            })
+        } else {
+            None
+        };
+
+        // Updation of Customer Details for the cases where both customer_id and specific customer
+        // details are provided in Payment Create Request
+        let key = key_store.key.get_inner().peek();
+        let customer_details = if raw_customer_details.is_some() {
+            raw_customer_details
+                .as_ref()
+                .map(Encode::encode_to_value)
+                .transpose()
+                .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                    field_name: "customer_details",
+                })
+                .attach_printable("Unable to convert guest customer details to a value")?
+                .map(Secret::new)
+                .async_lift(|inner| encrypt_optional(inner, key))
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Unable to encrypt guest customer details")?
```

Comment:
```suggestion
                .change_context(errors::ApiErrorResponse::InternalServerError)
                .attach_printable("Unable to encrypt customer details")?
```

Reply:
can you use this function https://github.com/juspay/hyperswitch/blob/185398a011d2a97cc84d0533605e118fed563d22/crates/router/src/core/payment_methods/cards.rs#L4350